### ISSUE TITLE
Flat symbol list C#

### ISSUE THREAD

**Issue by mattliewer** [patrys/vscode-code-outline#47]
In the latest version, for C# files, all symbols show up as a flat list.

![flathierarchy](https://user-images.githubusercontent.com/28793889/31364736-94aa0a82-ad1b-11e7-906f-5872cb93184f.png)

**Comment by patrys** [comment]
I guess C# reports symbols differently to most other languages. Unfortunately previous approach broke in many more languages so I think the best solution would be to file a bug against the C# symbol provider and have it report symbol ranges containing the entire class/method definition instead of just their names/signatures.

**Comment by mattliewer** [comment]
Ok, that makes sense. Any chance of providing an option in the settings to use the old method? I know that would not be ideal, but I miss having this work. :)

**Comment by christian-putzke** [comment]
I would really like to see the fallback option as well. Is there a plan to implement a fallback for C# through the settings?

**Comment by patrys** [comment]
No, sorry. The old method really did not work well for any languages.

**Comment by rakkarage** [comment]
https://github.com/OmniSharp/omnisharp-vscode/issues/2329

**Comment by longtth** [comment]
is the `code outline` in your image is the same with `outline` 
I have the lastest version of VSC but it can't parse my C# files. 
[Imgur](https://i.imgur.com/VWNb0e6.png)

**Comment by mattliewer** [comment]
@longtth I see the same behavior both in the `code outline` extension and the new built in `outline` functionality that ships with VS Code. If I am understanding your question correctly.

**Comment by longtth** [comment]
@mattliewer  : as you can see here [Imgur](https://i.imgur.com/m0OYwH9.png), the VSC 1.24.1 can't outline my Csharp file.
